### PR TITLE
Move components documentation to kyma docs

### LINKED ISSUE #979: Publish documentation how to install selected components
**Description**
Move the section "Install selected components only" from installer README.md to kyma docs.

**Links**
https://github.com/kyma-project/kyma/blob/master/components/installer/README.md#install-selected-components-only

### PR DESCRIPTION
**Pull request by strekm**
**Description**

Changes proposed in this pull request:

- Move components documentation to kyma docs

**Related issue(s)**
Resolve #979

### CODE REVIEW

File: docs/kyma/docs/040-cr-installation.md [hunk 1]
```diff
@@ -37,3 +37,23 @@ This table lists all the possible parameters of a given resource together with t
 | **metadata.finalizers** | **NO** | Protects the CR from deletion. Read [this](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers) Kubernetes document to learn more about finalizers. |
 | **spec.version** | **NO** | When manually installing Kyma on a cluster, specify any valid [SemVer](https://semver.org/) notation string. |
 | **spec.url** | **YES** | Specifies the location of the Kyma sources `tar.gz` package. For example, for the `master` branch of Kyma, the address is `https://github.com/kyma-project/kyma/archive/master.tar.gz` |
+
+## Install selected components only
```
Comment (pbochynski) [inline docs/kyma/docs/040-cr-installation.md]: Please read the whole chapter once again. Move the component list to first section. Make this description more consistent. What does "This tool" in this context?

File: docs/kyma/docs/040-cr-installation.md [hunk 2]
```diff
@@ -37,3 +37,23 @@ This table lists all the possible parameters of a given resource together with t
...
+
+This tool installs components specified in `Installation` CR. See the [installer-cr.yaml.tpl](../../../installation/resources/installer-cr.yaml.tpl) file for more details. 
+
```
Comment (kazydek) [inline docs/kyma/docs/040-cr-installation.md]: As for the link, currently, internal links do not work on Console UI and on the website so you can either remove it (and rephrase the description) or put the example of a file in the `assets` folder under the `kyma` directory in docs then the link to it would work but not sure how the example would be maintained later on if its definition changes?

File: docs/kyma/docs/040-cr-installation.md [hunk 3]
```diff
@@ -37,3 +49,7 @@ This table lists all the possible parameters of a given resource together with t
 | **metadata.finalizers** | **NO** | Protects the CR from deletion. Read [this](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers) Kubernetes document to learn more about finalizers. |
 | **spec.version** | **NO** | When manually installing Kyma on a cluster, specify any valid [SemVer](https://semver.org/) notation string. |
 | **spec.url** | **YES** | Specifies the location of the Kyma sources `tar.gz` package. For example, for the `master` branch of Kyma, the address is `https://github.com/kyma-project/kyma/archive/master.tar.gz` |
+| **spec.components** | **YES** | List of Helm charts components to `install` or `update` |
+| **spec.components.name** | **YES** | The name of the component is also the name of the component subdirectory in the `resources` directory. |
+| **spec.components.namespace** | **YES** | Namespace of the component in which it will be installed or updated. |
+| **spec.components.release** | **NO** | The name of release in case it should differ from component name. |
```
Comment (derberg) [inline docs/kyma/docs/040-cr-installation.md]: I don't think it is obvious where I can take this value from, I assume it is helm release name

File: docs/kyma/docs/040-cr-installation.md [hunk 4]
```diff
@@ -37,3 +49,7 @@ This table lists all the possible parameters of a given resource together with t
 | **metadata.finalizers** | **NO** | Protects the CR from deletion. Read [this](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers) Kubernetes document to learn more about finalizers. |
```
Comment (kazydek) [inline docs/kyma/docs/040-cr-installation.md]: As discussed, pls add the third option `update` in the **metadata.labels.action** field description.

Comment (strekm) [reply docs/kyma/docs/040-cr-installation.md]: done

File: docs/kyma/docs/040-cr-installation.md [hunk 5]
```diff
@@ -26,14 +26,30 @@ metadata:
...
+| **metadata.labels.action** | **YES** | Defines the behavior of the Kyma installer. Available options are `install`, `update` and `uninstall`. |
 | **metadata.finalizers** | **NO** | Protects the CR from deletion. Read [this](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers) Kubernetes document to learn more about finalizers. |
 | **spec.version** | **NO** | When manually installing Kyma on a cluster, specify any valid [SemVer](https://semver.org/) notation string. |
 | **spec.url** | **YES** | Specifies the location of the Kyma sources `tar.gz` package. For example, for the `master` branch of Kyma, the address is `https://github.com/kyma-project/kyma/archive/master.tar.gz` |
+| **spec.components** | **YES** | List of Helm charts components to `install` or `update` |
+| **spec.components.name** | **YES** | The name of the component is also the name of the component subdirectory in the `resources` directory. |
+| **spec.components.namespace** | **YES** | Namespace of the component in which it will be installed or updated. |
```
Comment (kazydek) [inline docs/kyma/docs/040-cr-installation.md]: `Defines the Namespace in which you want the Installer to install, uninstall, or update the component. `

File: docs/kyma/docs/040-cr-installation.md [hunk 6]
```diff
@@ -26,14 +26,30 @@ metadata:
...
+| **spec.components.release** | **NO** | The name of Helm release in case it should differ from component name. |
```
Comment (kazydek) [inline docs/kyma/docs/040-cr-installation.md]: `Provides the name of the Helm release in case it should differ from the component name.`

File: docs/kyma/docs/040-cr-installation.md [hunk 7]
```diff
@@ -26,14 +26,30 @@ metadata:
...
 
 | Field   |      Mandatory?      |  Description |
 |:----------:|:-------------:|:------|
 | **metadata.name** | **YES** | Specifies the name of the CR. |
-| **metadata.labels.action** | **YES** | Defines the behavior of the Kyma installer. Available options are `install` and `uninstall`. |
+| **metadata.labels.action** | **YES** | Defines the behavior of the Kyma installer. Available options are `install`, `update` and `uninstall`. |
 | **metadata.finalizers** | **NO** | Protects the CR from deletion. Read [this](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers) Kubernetes document to learn more about finalizers. |
 | **spec.version** | **NO** | When manually installing Kyma on a cluster, specify any valid [SemVer](https://semver.org/) notation string. |
 | **spec.url** | **YES** | Specifies the location of the Kyma sources `tar.gz` package. For example, for the `master` branch of Kyma, the address is `https://github.com/kyma-project/kyma/archive/master.tar.gz` |
+| **spec.components** | **YES** | List of Helm charts components to `install` or `update` |
```
[the remark this answers is not in the document]
Comment (strekm) [reply docs/kyma/docs/040-cr-installation.md]: i can make it `Lists which components of Helm chart components to `install` or `update`.` cause uninstall is still not supported in this scenario (uninstall is still hardcoded in installer itself, won't read components list)

File: docs/kyma/docs/040-cr-installation.md (hunk 5 above)
Comment (strekm) [reply docs/kyma/docs/040-cr-installation.md]: i can make it Defines the Namespace in which you want the Installer to `install`, or `update` the component. for the same reason: namespaces is irelevant when it comes to 'uninstall' action

File: docs/kyma/docs/040-cr-installation.md [hunk 8]
```diff
@@ -12,7 +12,7 @@ kubectl get crd installations.installer.kyma-project.io -o yaml
 
 ## Sample Custom Resource
 
-This is a sample CR that controls the Kyma installer. This example has the **action** label set to `install`, which means that it triggers the installation of Kyma.
+This is a sample CR that controls the Kyma installer. This example has the **action** label set to `install`, which means that it triggers the installation of Kyma. The  **name** and **namespace**  fields in the `components` array define which components you install and Namespaces in which you install them. This example shows that you install all components except for `remote-environments`. If you want the release name to be different from the component's name, provide the release parameter.
```
Comment (pbochynski) [inline docs/kyma/docs/040-cr-installation.md]: This is dangerous: "install all components except for `remote-environments`"
Please describe the convention used instead. It is the folder name in the resources folder from the URL or injected sources.

Comment (strekm) [reply docs/kyma/docs/040-cr-installation.md]: i guess that's already explained in the table. But would it be better to come back to previous example:
```
    - name: "remote-environments"
      namespace: "kyma-integration"
      release: "hmc-default"
```

File: docs/kyma/docs/040-cr-installation.md [hunk 9]
```diff
@@ -26,14 +26,24 @@ metadata:
 spec:
   version: "1.0.0"
   url: "https://sample.url.com/kyma_release.tar.gz"
+  components: 
+    - name: "remote-environments"
+      namespace: "kyma-integration"
```
[the remark this answers is not in the document]
Comment (strekm) [reply docs/kyma/docs/040-cr-installation.md]: i was trying to add generic, kyma-style example showing all "component features" at all and not putting whole real list of components here (cause as you pointed out it will be outdated at some point). There's note saying where full and up to date list can be found (please note that we can not use linking to our doc). For more detailed examples, use-cases, how-to's we should have separate docu explaining them in details (e.g. updating kyma-installation, adding  new component and so on)

File: docs/kyma/docs/040-cr-installation.md [hunk 10]
```diff
@@ -26,14 +28,24 @@ metadata:
...
 | **metadata.finalizers** | **NO** | Protects the CR from deletion. Read [this](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#finalizers) Kubernetes document to learn more about finalizers. |
 | **spec.version** | **NO** | When manually installing Kyma on a cluster, specify any valid [SemVer](https://semver.org/) notation string. |
 | **spec.url** | **YES** | Specifies the location of the Kyma sources `tar.gz` package. For example, for the `master` branch of Kyma, the address is `https://github.com/kyma-project/kyma/archive/master.tar.gz` |
+| **spec.components** | **YES** | Lists which components of Helm chart components to `install` or `update`. |
+| **spec.components.name** | **YES** | Specifies the name of the component which is the same as the name of the component subdirectory in the `resources` directory. |
+| **spec.components.namespace** | **YES** | Defines the Namespace in which you want the Installer to `install`, or `update` the component. |
+| **spec.components.release** | **NO** | Provides the name of the Helm release in case it should differ from the component name. |
```
Comment (Tomasz-Smelcerz-SAP) [inline docs/kyma/docs/040-cr-installation.md]: I would rephrase it like: "Provides the name of the Helm release. Defaults to component name."

Comment (kazydek) [reply docs/kyma/docs/040-cr-installation.md]: "Provides the name of the Helm release. The default one is the component name."
